src/client/cursor.rs

```diff
@@ -0,0 +1,160 @@
+use bson;
+use client::wire_protocol::flags::OpQueryFlags;
+use client::wire_protocol::operations::Message;
+use std::collections::vec_deque::VecDeque;
+use std::io::{Read, Write};
+
+/// Maintains a connection to the server and lazily returns documents from a
+/// query.
+///
+/// # Fields
+///
+/// `request_id` - Uniquely identifies the request being sent.
+/// `full_collection_name` - The full qualified name of the collection,
+///                          beginning with the database name and a period.
+/// `batch_size` - How many documents to fetch at a given time from the server.
+/// `cursor_id` - Uniquely identifies the cursor being returned by the reply.
+pub struct Cursor<'a, T> where T: Read + Write + 'a {
+    request_id: i32,
+    full_collection_name: String,
```

Comment:
Please call this `namespace` to remain consistent. `Collection` has a `namespace` field.